pkg/gofr/datasource/file/ftp/logger.go

```diff
@@ -0,0 +1,114 @@
+package ftp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// Level represents different logging levels.
+type Level int
+
+const (
+	DEBUG Level = iota + 1
+	INFO
+	ERROR
+)
+
+// FileLog handles logging with different levels.
+type FileLog struct {
+	Level     Level
+	Operation string    `json:"operation"`
+	Status    string    `json:"status,omitempty"`
+	Message   string    `json:"message,omitempty"`
+	Out       io.Writer `json:"out,omitempty"`
+	ErrOut    io.Writer `json:"err_out,omitempty"`
+}
+
+// NewLogger creates a new FileLog instance.
+func NewLogger(level Level) Logger {
+	return &FileLog{
+		Level:  level,
+		Out:    os.Stdout,
+		ErrOut: os.Stderr,
+	}
+}
+
+// clean cleans up the query string.
+func clean(query string) string {
+	query = regexp.MustCompile(`\s+`).ReplaceAllString(query, " ")
```

Comment:
`MustCompile` should be only used during initialization ( https://go.dev/wiki/CodeReviewComments#dont-panic ). Since this regexp is call invariant anyways, it should be compiled at init time, to avoid recompiling the regexp every time:
```
var regexpSpaces = regexp.MustCompile(`\s+`)
func clean(query string) string {
	return strings.TrimSpace(regexpSpaces.ReplaceAllString(query, " "))
}
```